src/table/utils/copy-utils.ts

```diff
@@ -1,9 +1,15 @@
 const copyCellValue = async (evt: any) => {
+  let value: string | undefined | null = '';
   const target = evt.target as HTMLElement;
 
-  const value = target.children[0]?.firstChild
-    ? target.children[0].firstChild.textContent
-    : target.firstChild?.textContent;
+  if (target.classList.contains('sn-table-head-cell')) {
+    value = target.children[0].firstChild?.textContent;
```

Comment:
This seem to be dependent on a very specific dom structure. Which adds the risk of making it break when that dom structure changes.

Possible risk mitigation for that could be to wrap the text in it's own span element that can then more easily be identifed?

Reply:
Just some spontaneous thinking here but something like this:
```js
if (target.tagName === 'svg') {
 target = target.parentNode;
}

value = target.querySelector('.span-with-text').textContent;
```

Reply:
yupp very good point. I was expecting this to break when we made changes, which we will soon.

That is a good suggestion. We would still need some logic to identify the parent then search for that span, but would be more general and robust. Not sure how much work it is to add an extra span though

Reply:
yeah it would be something like
```
if (!target.className.includes('sn-table-cell')) {
 target = target.parentNode; // or maybe grandparent, or search up don't remember the 
}

value = target.querySelector('.span-with-text').textContent;
```

Reply:
Maybe you can use [closest()](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest) when you need to traverse up the tree?

Reply:
should work

Reply:
I tried wrapping all type of cells in spans, seems to work out of the box. Let's go for this approach, much more robust. Would be good to do some sanity checking to see that the screen reader behaves  the same

Reply:
Will you also simplify the if logic as talked about?

Reply:
yes donya is working on that